index.js

```diff
@@ -408,13 +408,16 @@
     return BinaryType(name, functionUrl(name), test, _1, _2);
   }
 
+  //  applyParameterizedType :: (Type... -> Type) -> Type
+  function applyParameterizedType(type) {
+    return typeof type === 'function' ?
+      type.apply(null, Z.map(K(Unknown), range(0, type.length))) :
+      type;
+  }
+
```

Comment:
I'd prefer not to include this refactoring to keep the diff as clear as possible. If you explained your reasons for making this its own function in another comment, I haven't been able to find the comment. ;)